pytorch_lightning/utilities/migration.py

```diff
@@ -53,5 +49,6 @@ def __exit__(
     ) -> None:
         if hasattr(pytorch_lightning.utilities.argparse, "_gpus_arg_default"):
             delattr(pytorch_lightning.utilities.argparse, "_gpus_arg_default")
-        del sys.modules["pytorch_lightning.utilities.argparse_utils"]
-        _lock.release()
+
+        if "pytorch_lightning.utilities.argparse_utils" in sys.modules:
+            del sys.modules["pytorch_lightning.utilities.argparse_utils"]
```

Comment:
Is this the fix you are proposing?

Reply:
Yes, as far as I can tell this simple check should fix both the previous issue with multithreading and this issue.